internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -396,3 +398,96 @@ func (q amsQuotaService) GetSubscriptionByID(subscriptionID string) (*amsv1.Subs
 
 	return resp.Body(), true, nil
 }
+
+// Checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
+func (q amsQuotaService) IsQuotaEntitlementActive(kafka *dbapi.KafkaRequest) (bool, error) {
+	kafkaBillingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+	if err != nil {
+		return false, err
+	}
+
+	amsRelatedResourceBillingModel, err := q.getAMSRelatedResourceBillingModel(kafka.SubscriptionId, kafkaBillingModel)
+	if err != nil || shared.StringEmpty(amsRelatedResourceBillingModel) {
+		return false, err
+	}
+
+	orgID, err := q.amsClient.GetOrganisationIdFromExternalId(kafka.OrganisationId)
```

Comment:
side comment: I find extremely confusing the fact that `kafka.OrganisationId` is not an organisation id but an external id. It is even more confusing to see code like `GetOrganisationIdFromExternalId(kafka.OrganisationId)`.

I think we should evaluate the idea of renaming `OrganisationId` to `ExternalID`

Reply:
yeah got caught with that the first time round i was doing this check, I had it in my mind that the `kafka.OrganisationId` was the actual org id. +1 to revising this in the future.